sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2161,6 +2161,13 @@ object SQLConf {
     .booleanConf
     .createWithDefault(true)
 
+  val LEGACY_MSSQLSERVER_NUMERIC_MAPPING_ENABLED =
+    buildConf("spark.sql.legacy.mssqlserver.numericMapping.enabled")
+      .internal()
+      .doc("When true, use legacy MySqlServer SMALLINT and REAL type mapping.")
```

Comment:
It is not very clear what is the legacy MySqlServer SMALLINT and REAL type mapping if they do not read the PR. Could we make it clear? The behavior when the value is set to true and the behavior when the value is false

Reply:
Of course, I'd like to mention it at the migration guide fully.